src/modules/functionality.js

```diff
@@ -1,70 +1,73 @@
-// Function to render tasks on the page
-const renderTasks = (tasks, taskList) => {
-  // Clear existing tasks
-  taskList.innerHTML = '';
+/// RENDER TASKS///
+
+const taskList = document.getElementById('taskList');
 
+const renderTasks = () => {
+  const toDoArray = JSON.parse(localStorage.getItem('toDoArray')) || [];
+  taskList.innerHTML = '';
   // Loop through tasks and create taskBox element for each task
-  tasks.forEach((taskObject) => {
+  toDoArray.forEach((task) => {
     const taskBox = document.createElement('div');
     taskBox.classList = 'task-box-css';
     taskBox.innerHTML = ` <div class="task-activity">
                     <input type="checkbox" class="input-check">
-                    <input type="text" class="text-input" value="${taskObject.description}">
+                    <input type="text" class="text-input" value="${task.description}">
                   </div>
                   <i class="trash-icon" id="delete-icon">&#x1F5D1;</i>
                   <i class="move-icon" id="order-icon">&#x22EE;</i>`;
-    taskBox.dataset.taskObject = JSON.stringify(taskObject);
     taskList.appendChild(taskBox);
+
+    // COMPLETED TASKS//
+    const checkbox = taskBox.querySelector('.input-check');
+    checkbox.addEventListener('click', (e) => {
+      const taskBox = e.target.parentElement.parentElement;
+      const taskIndex = Array.from(taskBox.parentElement.children).indexOf(taskBox);
+      const toDoArray = JSON.parse(localStorage.getItem('toDoArray')) || [];
+      toDoArray[taskIndex].completed = e.target.checked;
```

Comment:
- [ ] Currently your updated array of items is stored in local storage but the correct list of values is not being reflected on the user interface after the page reloads. Kindly see my video below for more explanation.


https://www.loom.com/share/6eacc3e42fc84c6aa37f54bd34f926c1

Reply:
@topeogunleye I missed that one! sorry! I'll fix it right away! thanks for taking the time to record a video to show me!